dsl/Prop.cc

```diff
@@ -151,9 +173,16 @@ optional<NodesAndPropInfo> processProp(core::MutableContext ctx, ast::Send *send
     // From this point, we can't `return std::nullopt` anymore since we're going to be consuming the tree.
 
     NodesAndPropInfo ret;
+    ret.propInfo.name = name;
+    ret.propInfo.type = ASTUtil::dupType(type.get());
+    ret.propInfo.optional = isTNilable(type.get());
 
     // Compute the getters
     if (rules) {
+        if (ASTUtil::hasHashValue(ctx, *rules, core::Names::default_()) ||
+            ASTUtil::hasTruthyHashValue(ctx, *rules, core::Names::factory())) {
+            ret.propInfo.optional = true;
+        }
```

Comment:
Can we populate this information somewhere else? This is under the `// Compute the getters` section and more generally the `we're going to be consuming the tree` section. It seems like these operations are read-only.

Reply:
Sure. The only reason I put it in here to not introduce another branch on whether `rules` is `nullptr` or not. Should I just not bother trying to optimize a single branch instruction away?